Show errors from the tasks in the thread pool

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#33938 Show errors from the tasks in the thread pool**

Summary:
Making sure we don't silently ignore exceptions from the tasks in the
thread pool

Test Plan:
python setup.py clean && python setup.py develop install

Differential Revision: [D20178603](https://our.internmc.facebook.com/intern/diff/D20178603)

Commit:
Show errors from the tasks in the thread pool

Summary:
Making sure we don't silently ignore exceptions from the tasks in the
thread pool

Test Plan:
python setup.py clean && python setup.py develop install

[ghstack-poisoned]

dzhulgakov commented on c10/core/thread_pool.cpp:
```diff
@@ -108,7 +110,10 @@ void ThreadPool::main_loop(std::size_t index) {
         }
-      } catch (const std::exception&) {
+      } catch (const std::exception& e) {
+        C10_LOG_EVERY_MS(ERROR, 1000) << "Exception in thread pool task: " << e.what();
```
this usually shouldn't happen, so even regular LOG(ERROR) is good

Commit:
Update on "Show errors from the tasks in the thread pool"


Summary:
Making sure we don't silently ignore exceptions from the tasks in the
thread pool

Test Plan:
python setup.py clean && python setup.py develop install

[ghstack-poisoned]
